### ISSUE TITLE
Check avatar can generate thumbs correctly before adding avatar to db

### ISSUE THREAD

**Issue by spookyuser** [jazzband/django-avatar#171]
When an avatar is uploaded, maybe django avatar should first try get the avatar to make sure it uploaded correctly and only then write it to the db.

**Comment by spookyUnknownUser** [comment]
Hey 👋 
I think this is related to #153 

But I just stepped through the code and it seems like pngs with alpha pass this condition

https://github.com/grantmcconnaughey/django-avatar/blob/0d761fdd88710942217a58c4487e237f12916b53/avatar/models.py#L171-L172

The image mode isn't converted to RGB (maybe for valid reasons) and when we try save the avatar, if you go all the way through, pillow raises the  `cannot write mode %s as JPEG` error and then this bubbles up as a silent error over here

https://github.com/grantmcconnaughey/django-avatar/blob/0d761fdd88710942217a58c4487e237f12916b53/avatar/models.py#L179-L181

So I guess this can be pretty easily fixed by changing the default thumb storage to PNG. But I think maybe we should consider changing two things: 

(1) When PNGs have alpha and the thumb save format is JPG we should convert the image mode to RGB. This would stop the IOError

(2) Change the flow of the package a little. Instead of `User uploads file >Save file > Generate Thumbs` maybe we should do

````
User uploads file
Save file
Try generate thumbs
    if thumbs aren't generated 
        raise an error on the upload form
   else 
      add avatar to db
````

That way if another error like this happens, the avatar won't be added to the db, the user will see an error and broken links won't be displayed instead of their avatar.

I would be happy to try make a PR that does this, if this makes sense?

**Comment by johanneswilm** [comment]
This has been fixed in version 6.0.0.
